### PR TITLE
Storm auth command updates (SYN-5240, SYN-5265)

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Add index arguments for adding rules (SYN-5265)
- Add gate admin status to auth.user.show (SYN-5240)

### CODE REVIEW

File: synapse/lib/stormtypes.py [hunk 1]
```diff
@@ -8068,7 +8068,9 @@ class User(Prim):
          'type': {'type': 'function', '_funcname': '_methUserAddRule',
                   'args': (
                       {'name': 'rule', 'type': 'list', 'desc': 'The rule tuple to add to the User.', },
-                      {'name': 'gateiden', 'type': 'str', 'desc': 'The gate iden used for the rule.', 'default': None, }
+                      {'name': 'gateiden', 'type': 'str', 'desc': 'The gate iden used for the rule.',
+                       'default': None, },
+                      {'name': 'indx', 'type': 'int', 'desc': 'The index to add the rule in.', 'default': None, }
```
Comment (invisig0th) [inline synapse/lib/stormtypes.py:8073]: desc is a bit awkwardly phrased here and below. I like the desc from above. Perhaps something like `The position of the rule as a 0 based index.`

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/stormlib/auth.py [hunk 2]
```diff
@@ -201,17 +201,21 @@
 
                 // add a deny rule to the user "visi" for permission "foo.bar.baz"
                 auth.user.addrule visi "!foo.bar.baz"
+
+                // add a allow rule to the user "visi" for permission "baz" at the first index.
```
Comment (Cisphyx) [inline synapse/lib/stormlib/auth.py:205]: "an" allow rule

File: synapse/lib/stormlib/auth.py [hunk 3]
```diff
@@ -269,17 +273,21 @@
 
                 // add a deny rule to the role "ninjas" for permission "foo.bar.baz"
                 auth.role.addrule ninjas "!foo.bar.baz"
+
+                // add a allow rule to the user "visi" for permission "baz" at the first index.
```
Comment (Cisphyx) [inline synapse/lib/stormlib/auth.py:277]: "an" allow rule
also, for the role "ninjas" not a user

**Review by invisig0th — COMMENTED** [review]
Once cisphyx's changes are done, i think this is good to go :+1:

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**